packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -214,56 +288,52 @@ export const GridRootStyles = styled('div', {
       // to anchor the aggregation label
       position: 'relative',
     },
-    [`& .${gridClasses.columnHeaderTitleContainerContent}`]: {
+    [`& .${c.columnHeaderTitleContainerContent}`]: {
       overflow: 'hidden',
       display: 'flex',
       alignItems: 'center',
     },
-    [`& .${gridClasses['columnHeader--filledGroup']} .${gridClasses.columnHeaderTitleContainer}`]: {
+    [`& .${c['columnHeader--filledGroup']} .${c.columnHeaderTitleContainer}`]: {
       borderBottomWidth: '1px',
       borderBottomStyle: 'solid',
       boxSizing: 'border-box',
     },
-    [`& .${gridClasses['columnHeader--filledGroup']}.${gridClasses['columnHeader--showColumnBorder']} .${gridClasses.columnHeaderTitleContainer}`]:
+    [`& .${c['columnHeader--filledGroup']}.${c['columnHeader--showColumnBorder']} .${c.columnHeaderTitleContainer}`]:
       {
         borderBottom: `none`,
       },
-    [`& .${gridClasses['columnHeader--filledGroup']}.${gridClasses['columnHeader--showColumnBorder']}`]:
-      {
-        borderBottomWidth: '1px',
-        borderBottomStyle: 'solid',
-        boxSizing: 'border-box',
-      },
-    [`& .${gridClasses.headerFilterRow}`]: {
-      borderTop: `1px solid ${borderColor}`,
```

Comment:
The border is missing between the headers and header filters:
https://app.argos-ci.com/mui/mui-x/builds/16834/70974514

Reply:
Ah, it's applied to the cells now, this is why it's not there when custom cells are used.

Reply:
Should we fix this demo separately then?

Reply:
Yes, it's not a blocker

Reply:
[Fixed](https://github.com/mui/mui-x/pull/10059/commits/11c84c5b80b2890caa5f6c27bf66fa0575dea88f)